api/api_keys/models.py

```diff
@@ -18,3 +20,12 @@ class MasterAPIKey(AbstractAPIKey, SoftDeleteObject):
 
     objects = MasterAPIKeyManager()
     is_admin = models.BooleanField(default=True)
+
+    @hook(BEFORE_UPDATE, when="is_admin", was=False, is_now=True)
+    def delete_role_api_keys(
```

Comment:
We should add a test for this method.

Reply:
The test is here: https://github.com/Flagsmith/flagsmith-rbac/pull/19

Reply:
In that case, I think we should move all of this code to a signal in the `RBAC` repository.